### PR TITLE
feat(snyk-ls): add support for snyk-ls

### LINKED ISSUE #2825: snyk-ls
### Language server

snyk-ls

### Requested feature

Add configuration for [snyk-ls](https://github.com/snyk/snyk-ls)

### Other clients which have this feature

_No response_

### PR DESCRIPTION
**Pull request by pbnj**
Closes #2825

TODO:
- [ ] Add docs

### CODE REVIEW

File: lua/lspconfig/server_configurations/snyk_ls.lua [hunk 1]
```diff
@@ -0,0 +1,24 @@
+local util = require 'lspconfig.util'
+
+return {
+  default_config = {
+    cmd = { 'snyk-ls' },
+    root_dir = util.find_git_ancestor,
+    single_file_support = true,
+    filetypes = {},
```
Comment (glepnir) [inline lua/lspconfig/server_configurations/snyk_ls.lua:8]: filetypes shouldn't empty

Comment (pbnj) [reply lua/lspconfig/server_configurations/snyk_ls.lua:8]: Snyk supports many languages, including infra-as-code languages (e.g. Terraform, Helm/Kubernetes manifests, AWS Cloudformation).

I looked at language-agnostic LSPs for inspiration, like [diagnostic-languageserver](https://github.com/neovim/nvim-lspconfig/blob/master/lua/lspconfig/server_configurations/diagnosticls.lua).

Is there a better way to handle this case, like `filetype = {"*"}`?

Or do you recommend specifying every filetype?

How do you handle something like AWS Cloudformation which could be JSON or YAML?

Comment (glepnir) [reply lua/lspconfig/server_configurations/snyk_ls.lua:8]: I think it's better remove `filetype` and add config example in `doc.describe`.

Comment (pbnj-dragon) [reply lua/lspconfig/server_configurations/snyk_ls.lua]: Done

Comment (pbnj) [reply lua/lspconfig/server_configurations/snyk_ls.lua]: Done

**Pull request merged**